ambry-router/src/main/java/com/github/ambry/router/QuotaAwareOperationController.java

```diff
@@ -252,8 +252,11 @@ private void pollQuotaExceedAllowedRequestsIfAny(List<RequestInfo> requestsToSen
 
   @Override
   protected List<ResponseInfo> getNonQuotaCompliantResponses() {
-    List<ResponseInfo> nonCompliantResponses = nonCompliantRequests.
-        stream().map(requestInfo -> new ResponseInfo(requestInfo, true)).collect(Collectors.toList());
+    List<ResponseInfo> nonCompliantResponses = nonCompliantRequests.stream().map(requestInfo -> {
+      // Release the Netty ByteBuf here since they are not sent out to Network layer.
+      requestInfo.getRequest().release();
```

Comment:
Good catch.
Btw, this means that a memory leak would happen only when requests are rejected. In the current implementation, rejection happens only due to storage quota.